wherehows-web/app/routes/search.ts

```diff
@@ -2,36 +2,33 @@ import Route from '@ember/routing/route';
 import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
 import createSearchEntries from 'wherehows-web/utils/datasets/create-search-entries';
 import { refreshModelQueryParams } from 'wherehows-web/utils/helpers/routes';
-import SearchController from 'wherehows-web/controllers/search';
-import { readSearch, ISearchApiParams } from 'wherehows-web/utils/api/search';
+import { readSearch } from 'wherehows-web/utils/api/search';
+import { action } from '@ember-decorators/object';
 
 export default class SearchRoute extends Route.extend(AuthenticatedRouteMixin) {
   // Set `refreshModel` for each queryParam to true
   //  so each url state change results in a full transition
-  queryParams = refreshModelQueryParams(['keyword', 'category', 'page', 'source']);
+  queryParams = refreshModelQueryParams(['category', 'page', 'facets', 'keyword']);
 
   /**
-   * Applies the returned results object as the route model and sets
-   *   keyword property on the route controller
-   * @param {Object} controller search route controller
-   * @param {Object} model search results
+   * Makes an API call and process search entries
    */
-  setupController(controller: SearchController, model: any) {
-    const { keywords } = model;
-
-    controller.setProperties({
-      model,
-      keyword: keywords
-    });
+  async model(apiParams: any) {
+    const { result } = await readSearch(apiParams);
+    const { keywords, data } = result || { keywords: '', data: [] };
+    createSearchEntries(data, keywords);
+    return result || {};
   }
 
   /**
-   * Makes an API call and process search entries
+   * Add spinner when model is loading
    */
-  async model(apiParams: ISearchApiParams) {
-    const { result } = await readSearch(apiParams);
-    const { keywords, data } = result;
-    createSearchEntries(data, keywords);
-    return result;
+  @action
+  loading(transition: any) {
```

Comment:
We're using ember-concurrency in wherehows, so you can actually remove the need for this by just showing a spinner on the template if the task is running.

Reply:
I know, but this is a little bit different. First, this is in a route, so we are using model to fetch (standard ember). And then, I am tweaking a little bit the behavior of loading, since it starts before the debounce happens :)